### ISSUE TITLE
formatResponse typings: null response?

### ISSUE THREAD

**Issue by nwalters512** [apollographql/apollo-server#5009]
In https://github.com/apollographql/apollo-server/pull/3431, the typings for `formatResponse` were set such that the `response` argument could be `null`. However, this causes problems when trying to write a basic implementation that, for instance, adds a property to the object:

```ts
new ApolloServer({
  formatResponse: (response) => {
    if (response) {
      if (!response.extensions) {
        response.extensions = {};
      }
      response.extensions.code_version = VERSION;
    }
    return response;
  }
})
```

This produces a TypeScript error, as `response` is typed as `GraphQLResponse | null` but the expected return type is just `GraphQLResponse` (non-nullable):

```
Type '(response: GraphQLResponse | null) => GraphQLResponse | null' is not assignable to type '(response: GraphQLResponse | null, requestContext: GraphQLRequestContext<object>) => GraphQLResponse'.
  Type 'GraphQLResponse | null' is not assignable to type 'GraphQLResponse'.
    Type 'null' is not assignable to type 'GraphQLResponse'.
```

It's not clear to me if `response` ever actually *can* be null here, but I would expect this function to be typed such that the `response` argument and the return value have the same type. If `response` will not ever be null, I'd like the types to reflect that.

This is mentioned in a comment on https://github.com/apollographql/apollo-server/issues/4186, though I don't think that's actually the issue that OP encountered there, so I opted to lift this into a new issue.

**Comment by andrewphillipo** [comment]
for some reason this works for me:

```es6
return assign({}, response, { extensions: { code_version: VERSION } });
```

**Comment by glasser** [comment]
Thanks for the report. Looks like we put the `| null` in the wrong place in #3431. Looks like #5089 should fix it.

**Comment by glasser** [comment]
I've released a prerelease with this fix, version `2.23.0-alpha.0`. Try out the alpha and see if it works for you! Please provide any feedback in #5094.

**Comment by glasser** [comment]
This is released in Apollo Server 2.23.0.
